### PR TITLE
DM-46471: cm-service cli update collections and update child config need to be explicitly passed

### PR DESCRIPTION
**Pull request by eigerx**

### CODE REVIEW

**Review by fritzm — APPROVED** [review]
LGTM -- thanks!

File: examples/start_hsc_micro.yaml
Comment (eacharles) [inline examples/start_hsc_micro.yaml:1]: it looks like a number of changes from your working version snuck into the PR in the start_hsc_micro.yaml file.   You probably want to back everything except the collection -> collections fix out.

File: src/lsst/cmservice/db/node.py [hunk 1]
```diff
@@ -246,6 +246,12 @@ async def resolve_collections(
                     raise CMResolveCollectionsError(
                         f"Failed to resolve collection {name_}, {f1} using: {name_dict!s}",
                     ) from msg
+        for key, value in resolved_collections:
```
Comment (eacharles) [inline src/lsst/cmservice/db/node.py:249]: this probably needs to be `for key, value in resolved_collections.itemsI()`

**Review by eacharles — CHANGES_REQUESTED** [review]

**Comment by eacharles** [comment]
You are also going to want to update tests/db/test_micro.py to 

1. Grab the campaign like so: campaign = await interface.load_and_create_campaign(...)
2. override the MUST_OVERRIDE stuff in the campaign.collections with  await campaign.update_collections(...)
3. overide the MUST_OVERRIDE stuff in the campaign.data with  await campaign.update_data_dict(...).   I'd recommend setting prod_area to something different that output/archive...  maybe output_test/archive
4. clean the prod_area with os.system('\rm -rf output_test/archive') at the end of the test.

File: examples/start_hsc_micro.yaml [hunk 2]
```diff
@@ -2,18 +2,16 @@ Production:
...
+    spec_name: hsc_micro_panda
+    collections:
         out: MUST_OVERRIDE
-        # campaign_source: HSC/raw/RC2
-        # campaign_ancillary_inputs: "HSC/calib,HSC/masks,HSC/fgcmcal/lut/RC2,refcats,skymaps"
+        campaign_source: HSC/raw/RC2
+        campaign_ancillary_inputs: "HSC/calib,HSC/masks,HSC/fgcmcal/lut/RC2,refcats,skymaps"
     data:
-        lsst_version: MUST_OVERRIDE
+        lsst_version: w_2024_38
```
Comment (eacharles) [inline examples/start_hsc_micro.yaml:11]: Please change this to MUST_OVERRIDE

File: examples/start_hsc_micro.yaml [hunk 3]
```diff
@@ -2,18 +2,16 @@ Production:
...
         prod_area: MUST_OVERRIDE
-#       lsst_version: w_2024_22
-#       prod_area: 'output/archive'
-#       butler_repo: '/repo/main'
-#       data_query: "instrument = 'HSC' AND exposure in (30504, 30502) AND detector in (45, 46, 47, 48)"
+        butler_repo: '/repo/main'
```
Comment (eacharles) [inline examples/start_hsc_micro.yaml:13]: I think it would be better to leave these commented.

**Review by eacharles — APPROVED** [review]
Looks good expect for a couple of things still outstanding in start_hsc_micro.yaml

**Pull request merged**